components/bm1397/bm1397.c

```diff
@@ -12,6 +12,28 @@
 #include "bm1397.h"
 #include "utils.h"
 #include "crc.h"
+#include "mining.h"
+#include "global_state.h"
```

Comment:
bm1397 shouldnt have to import global_state.h

Reply:
It needs to refrence the global state struct for BM1397_send_work etc.

Reply:
Yeah that likely means there is something wrong with the api then.